modules/DMZ/models/classification/digit_recognition

```diff
@@ -0,0 +1,99 @@
+                img_array.append(skimage.io.imread(str_path + ext)) #add image to img_array
+                target_str = str_path[path_length:]  # build the target array to correspond with image collection
+                target_str = target_str[:7]
+                target_int = int(target_str[-1:]) #get the number that the img represents from the filename
+                target_array.append(target_int) #append to target_array
+
+        #Build the matrix of rows x features
+        n_samples = len(img_array)
+        features = 28 * 28  # size of image
+        row_array = []
+        matrix = []
+
+        for member in (img_array):  # build the matrix
+            row_array = numpy.reshape(member, (features))
+            matrix.append(row_array)
+
+        #randomize the data
+        a = tuple(matrix)
+        b = tuple(target_array)
+        zip_list = list(zip(a,b))      # tie each entry in the matrix to its prediction value
+        numpy.random.shuffle(zip_list) # randomize
+
+        train_int = int(0.7 * n_samples)  #amt of data for training
+        test_int = n_samples - train_int  #amt of data for testing
+
+        digit_recognition.train = zip_list[:test_int] #70% of the data stored in train_data
+        digit_recognition.test  = zip_list[-test_int:] #30% of the data stored in test_data
+
+        #return a tuple of zipped lists
+        #return train_data, test_data
+
+
+    # Given a data in the form of a tuple of two zipped lists (each image is paired with its classification)
+    # and each list corresponds to testing & training data,
+    # This function creates a SVM classifier, fits the data using the training data
+    # and returns a tuple of two lists - one list is the prediction, one is the actual value.
+    def predict_images(self):
+        train_data, test_data = digit_recognition.train, digit_recognition.test   # get testing & training data
+        matrix_train_tup, digits_train_tup = zip(*train_data)  # unzip the randomized training list
+        matrix_train = list(matrix_train_tup)
+        digits_train = list(digits_train_tup)
+
+        # Create a classifier: a support vector classifier
+        classifier = svm.SVC(C=1.0, tol=1e-10, cache_size=600, kernel='rbf', gamma=0.00000001)
+
+        # Classify the Training Data with the fit() function
+        classifier.fit(matrix_train, digits_train)
+
+        # Get the prediction list using the testing data
+        matrix_test_tup, digits_test_tup = zip(*test_data)  # unzip the randomized testing list
+        matrix_test = list(matrix_test_tup)
+        digits_test = list(digits_test_tup)
+
+        #get expected and predicted values in list form
+        expected = digits_test
+        predicted = list(classifier.predict(matrix_test))
+
+        #return a tuple of two lists
+        return predicted, expected
+
+#Testing
+path = "/mnt/c/Users/Tayo/Documents/SE2CS4365/HandWrittenChar/eggChar/mnist_jpgfiles/trainsmall/"
```

Comment:
This path will not work for anyone else. If we run this on our machines we will get an error because the path does not exist. Please do not use absolute pathing. We have code [here](https://github.com/ASAAR/SE2-KaggleComp/blob/master/setup.py) for relative paths in the project.